### PR TITLE
In Progress

### PR DESCRIPTION
**Pull request by pskoczenski**

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/elements/fandom/pages/HubPage.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  @Override
+  public HubPage open() {
+    getUrl(String.format("http://qa.fandom.wikia.com/%s", "games"));
+
+    return this;
+  }
+
+    @Getter(lazy = true)
+    private final HeroUnit heroUnit = new HeroUnit();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/pages/HubPage.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/elements/fandom/pages/HubPage.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    private String URL = "http://qa.fandom.wikia.com/";
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/pages/HubPage.java]: MAJOR Rename this field "URL" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)
MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 3 issues:
* MAJOR 1 major
* MINOR 2 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/elements/fandom/components/SearchInput.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
+package com.wikia.webdriver.elements.fandom.components;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+import org.apache.xpath.operations.String;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/SearchInput.java]: MINOR Remove this unused import 'org.apache.xpath.operations.String'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/fandom/components/SearchInput.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import org.openqa.selenium.By;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/SearchInput.java]: MINOR Remove this unused import 'org.openqa.selenium.By'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+/**
+ * Created by psko on 8/1/16.
+ */
+
+@Test(groups = {"Fandom", "Fandom_Search"})
+public class SearchInputTest extends FandomTestTemplate {
+
+    @Test
+    public void searchInputAcceptsValue() {
+        SearchInput searchInput = new SearchInput().searchInputValue();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Remove this unused "searchInput" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        SearchInput searchInputSubmit = new SearchInput().searchInputSubmit();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Remove this unused "searchInputSubmit" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 7]
```diff
@@ -0,0 +1,30 @@
+package com.wikia.webdriver.testcases.fandom;
+
+import com.gargoylesoftware.htmlunit.javascript.host.URL;
+import com.wikia.webdriver.common.core.Assertion;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MINOR Remove this unused import 'com.wikia.webdriver.common.core.Assertion'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+
+@Test(groups = {"Fandom", "Fandom_Search"})
+public class SearchInputTest extends FandomTestTemplate {
+
+    @Test
+    public void searchInputAcceptsValue() {
+        SearchInput searchInput = new SearchInput().searchInputValue();
+        SearchInput searchInputSubmit = new SearchInput().searchInputSubmit();
+
+        String URL = driver.getCurrentUrl();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00117)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+        Assert.assertEquals(URL, "http://qa.fandom.wikia.com/?s=TEST");
+
+        SearchInput firstInstance = new SearchInput().firstPostInstance();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Remove this unused "firstInstance" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 7 issues:
* MAJOR 4 major
* MINOR 3 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+
+    }
+
+    @Test
+    public void differentUrlForPostsAndVideo() {
+        SearchInput searchInput = new SearchInput().searchInputValue();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Remove this unused "searchInput" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+        SearchInput searchInputSubmit = new SearchInput().searchInputSubmit();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Remove this unused "searchInputSubmit" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+        String URL = driver.getCurrentUrl();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Rename this local variable name to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00117)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+        Assert.assertEquals(URL, "http://qa.fandom.wikia.com/?s=TEST");
+
+        SearchInput postsResult = new SearchInput().searchResultsPosts();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Remove this unused "postsResult" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 14]
```diff
@@ -0,0 +1,49 @@
...
+
+        String postUrl = driver.getCurrentUrl();
+
+        SearchInput videosResult = new SearchInput().searchResultsVideos();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Remove this unused "videosResult" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/elements/fandom/pages/HubPage.java (hunk 1 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/pages/HubPage.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)

File: src/test/java/com/wikia/webdriver/elements/fandom/pages/HubPage.java (hunk 2 above)
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/pages/HubPage.java]: MINOR Move this variable to comply with Java Code Conventions. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1213)
MAJOR Rename this field "URL" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS00116)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 15 issues:
* MAJOR 10 major
* MINOR 5 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 15]
```diff
@@ -0,0 +1,50 @@
+package com.wikia.webdriver.testcases.fandom;
+
+import com.gargoylesoftware.htmlunit.javascript.host.URL;
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.templates.fandom.FandomTestTemplate;
+import com.wikia.webdriver.elements.fandom.components.SearchInput;
+import org.openqa.selenium.By;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MINOR Remove this unused import 'org.openqa.selenium.By'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+
+        Assert.assertEquals(firstPostTitle, postTitleFinal);
+
+    }
+
+    @Test
+    public void differentUrlForPostsAndVideo() {
+        SearchInput searchInput = new SearchInput().searchInputValue();
+        SearchInput searchInputSubmit = new SearchInput().searchInputSubmit();
+        SearchInput postsResult = new SearchInput().searchResultsPosts();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Remove this unused "postsResult" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 17]
```diff
@@ -0,0 +1,50 @@
+package com.wikia.webdriver.testcases.fandom;
+
+import com.gargoylesoftware.htmlunit.javascript.host.URL;
+import com.wikia.webdriver.common.core.Assertion;
+import com.wikia.webdriver.common.templates.fandom.FandomTestTemplate;
+import com.wikia.webdriver.elements.fandom.components.SearchInput;
+import org.openqa.selenium.By;
+import org.openqa.selenium.support.FindBy;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MINOR Remove this unused import 'org.openqa.selenium.support.FindBy'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java [hunk 18]
```diff
@@ -0,0 +1,50 @@
...
+    @Test
+    public void searchInputAcceptsValue() {
+        SearchInput searchInput = new SearchInput().searchInputValue();
+        SearchInput searchInputSubmit = new SearchInput().searchInputSubmit();
+
+        String URL = driver.getCurrentUrl();
+        Assert.assertEquals(URL, "http://qa.fandom.wikia.com/?s=TEST");
+
+        SearchInput firstPostTitle = new SearchInput().getPostTitle();
+        SearchInput firstInstance = new SearchInput().firstPostInstance();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/SearchInputTest.java]: MAJOR Remove this unused "firstInstance" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 16 issues:
* MAJOR 9 major
* MINOR 7 minor

Watch the comments in this conversation to review them.

File: src/test/java/com/wikia/webdriver/testcases/fandom/LoadMoreTest.java [hunk 19]
```diff
@@ -0,0 +1,21 @@
...
+import com.wikia.webdriver.elements.fandom.components.LoadMore;
+import com.wikia.webdriver.elements.fandom.pages.MainPage;
+import org.testng.annotations.Test;
+
+@Test(groups = {"Fandom", "Fandom_Load_More"})
+public class LoadMoreTest extends FandomTestTemplate {
+
+    @Test
+    public void loadMoreButtonFunctionsCorrectly() {
+        MainPage mainPage = new MainPage().open();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/LoadMoreTest.java]: MAJOR Remove this unused "mainPage" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/testcases/fandom/LoadMoreTest.java [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+        LoadMore loadMore = new LoadMore().loadMoreBtn();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/LoadMoreTest.java]: MAJOR Remove this unused "loadMore" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/testcases/fandom/LoadMoreTest.java [hunk 21]
```diff
@@ -0,0 +1,21 @@
...
+        LoadMore topStories = new LoadMore().moreTopStories();
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/testcases/fandom/LoadMoreTest.java]: MAJOR Remove this unused "topStories" local variable. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AS1481)

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java [hunk 22]
```diff
@@ -0,0 +1,33 @@
+package com.wikia.webdriver.elements.fandom.components;
+
+import com.wikia.webdriver.pageobjectsfactory.pageobject.BasePageObject;
+import org.openqa.jetty.html.List;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: MINOR Remove this unused import 'org.openqa.jetty.html.List'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java [hunk 23]
```diff
@@ -0,0 +1,33 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindAll;
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/elements/fandom/components/LoadMore.java]: MINOR Remove this unused import 'org.openqa.selenium.support.FindAll'. [rule](http://sonar-s1/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 21 issues:
* MAJOR 12 major
* MINOR 9 minor

Watch the comments in this conversation to review them.

**Comment by nikodamn** [comment]
Is this PR still valid? Should we close it?

**Comment by pskoczenski** [comment]
It's no longer valid and can be closed.

**Comment by WikiaTech** [comment]
SonarQube analysis reported 21 issues:
* MAJOR 12 major
* MINOR 9 minor

Watch the comments in this conversation to review them.

**Pull request closed without merging**